lib/send_assets.js

```diff
@@ -7,25 +8,41 @@ const { convertCSVToArray } = require('convert-csv-to-array')
 const generateTransactions = require('../lib/generate_transaction_types')
 const helper = require('../lib/helper.js')
 const BigN = require('js-big-decimal')
+const events = require('./events')
 
 const SendAssets = {
-  getMailboxSendTransactions: async function (algodClient, assetId, amount, fromAccount, 
-    toAddress) {
-
-    const isOptedIn = process.env.NEXT_PUBLIC_ALWAYS_SEND_TO_ESCROW ? false :
-      await helper.checkOptIn(toAddress, assetId)
+  getMailboxSendTransactions: async function (
+    algodClient,
+    assetId,
+    amount,
+    fromAccount,
+    toAddress
+  ) {
+    const isOptedIn = process.env.NEXT_PUBLIC_ALWAYS_SEND_TO_ESCROW
+      ? false
+      : await helper.checkOptIn(toAddress, assetId)
     if (isOptedIn) {
-      return await generateTransactions.getSimpleSendAssetTxn(algodClient, 
-        assetId, amount, fromAccount, toAddress)
+      return await generateTransactions.getSimpleSendAssetTxn(
+        algodClient,
+        assetId,
+        amount,
+        fromAccount,
+        toAddress
+      )
     }
 
     return await generateTransactions.getFundEscrowTxns(
-      algodClient, assetId, amount, fromAccount, toAddress
+      algodClient,
+      assetId,
+      amount,
+      fromAccount,
+      toAddress
     )
   },
   send: async function (assetId, fromAddress, csvInput) {
     try {
       const { algodClient, feeReceiverAddress } = helper.getAlgodex()
+      events.emit('sending-assets-event', { status: 'loading' })
```

Comment:
- [ ] Unnecessary event, we only need events from inside of each step and not the overall progress.